src/components/network-area-diagram-viewer/diagram-utils.ts

```diff
@@ -622,3 +626,61 @@ function getElementType(element: SVGElement | null): ElementType {
     }
     return ElementType.UNKNOWN;
 }
+
+// get view box computed starting from node and text positions
+// defined in diagram metadata
+export function getViewBox(
+    nodes: NodeMetadata[] | undefined,
+    textNodes: TextNodeMetadata[] | undefined,
+    svgParameters: SvgParameters
+): ViewBox {
+    const size = { minX: Number.MAX_VALUE, maxX: -Number.MAX_VALUE, minY: Number.MAX_VALUE, maxY: -Number.MAX_VALUE };
+    const nodesMap: Map<string, NodeMetadata> = new Map<string, NodeMetadata>();
```

Comment:
I don't get why you need this map... besides it''s probably slowing down a bit the calculation

Reply:
The JSON metadata, for the text nodes, contains the shift w.r.t. vl node position, not the text node position. So when computing min and max for text nodes, besides the text node metadata, I need vl node positions. An option would be, for each vl node connected to a text node, to search it in the metadata. The other option is to save the position when computing min and max for vl nodes, so to have it available in a map.